api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -105,6 +109,87 @@ type ProxyLogging struct {
 	Level map[LogComponent]LogLevel `json:"level,omitempty"`
 }
 
+type ProxyAccessLogging struct {
+	// Format defines the format of access logging.
+	Format ProxyAccessLoggingFormat `json:"format"`
+	// Sink defines the sink of access logging.
+	Sink ProxyAccessLoggingSink `json:"sink"`
+}
+
+type ProxyAccessLoggingFormatType string
+
+const (
+	// ProxyAccessLoggingFormatTypeText defines the text access logging format.
+	ProxyAccessLoggingFormatTypeText ProxyAccessLoggingFormatType = "Text"
+	// ProxyAccessLoggingFormatTypeJSON defines the JSON access logging format.
+	ProxyAccessLoggingFormatTypeJSON ProxyAccessLoggingFormatType = "JSON"
+	// TODO: support format type "mix" in the future.
+)
+
+type ProxyAccessLoggingFormat struct {
+	// Type defines the type of access logging format.
+	// +kubebuilder:validation:Enum=Text;JSON
+	Type ProxyAccessLoggingFormatType `json:"type,omitempty"`
+	// Text defines the text access logging format, following Envoy access logging formatting,
+	// empty value results in proxy's default access log format.
+	// It's required when the format type is "text".
+	// Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators) may be used in the format.
+	// The [format string documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-strings) provides more information.
+	// +optional
+	Text *string `json:"text,omitempty"`
```

Comment:
What would be the semantic difference between nil and an empty string? It seems for both we would just use the default format and so the extra redirection from the pointer seems unnecessary.

Reply:
yes, there's no different now.
I'd like keep same as other properties in this project like https://github.com/envoyproxy/gateway/blob/d801f1c7c5152ac68e2c013e9d7f3eab13731d11/api/config/v1alpha1/shared_types.go#L111

https://github.com/envoyproxy/gateway/blob/d801f1c7c5152ac68e2c013e9d7f3eab13731d11/docs/latest/design/bootstrap.md?plain=1#L43